dfastmi/batch/AnalyserAndReporterWaqua.py

```diff
@@ -96,180 +86,282 @@ def analyse_and_report_waqua(
+        Arguments
+        ---------
+        fraction_of_year : Vector
+            Fraction of year represented by each forcing condition.
+        rsigma : Vector
+            Array of relaxation factors; one per forcing condition.
+
+        Returns
+        -------
+        success : bool
+            Flag indicating whether analysis could be carried out.
+        output_data : OutputDataWaqua
+            data used to generate a report.
+        """   
+        success, dzq, firstm, firstn = self._get_output_values()
+        output_data = self._calculate_output_data(fraction_of_year, rsigma, dzq, firstm, firstn)
+        return success, output_data
+
+    def _get_output_values(self) -> Tuple[bool, numpy.ndarray, int, int]:
+        success = True
+        first_discharge = True
+        dzq : List[Optional[Union[float, numpy.ndarray]]]
+        dzq = [None] * len(self.discharges)
+        for i in range(3):
+            if success and self.apply_q[i]:
+                if first_discharge:
+                    dzq[i], firstm, firstn = self._get_values_waqua(i+1, self.discharges[i])
+                    first_discharge = False
+                else:
+                    dzq[i], _, _ = self._get_values_waqua(i+1, self.discharges[i])
+                if dzq[i] is None:
+                    success = False
+            else:
+                dzq[i] = 0.0
+        return success, dzq, firstm, firstn
+
+    def _get_values_waqua(self, stage: int, discharge_value: float) -> Tuple[numpy.ndarray, int, int]:
+        """
+        Read data files exported from WAQUA for the specified stage, and return equilibrium bed level change and minimum M and N.
+
+        Arguments
+        ---------
+        stage : int
+            Discharge level (1, 2 or 3).
+        discharge_value : float
+            Discharge value.
+
+        Returns
+        -------
+        dzq : numpy.ndarray
+            Array containing equilibrium bed level change.
+        firstm : int
+            Minimum M index read (0 if reduced_output is False).
+        firstn : int
+            Minimum N index read (0 if reduced_output is False).
+        """
+        files_found, files = self._search_files(stage, discharge_value)
+        if not files_found:
             return numpy.array([]), 0, 0
-        else:
-            if display:
-                ApplicationSettingsHelper.log_text("input_xyz_found", dict={"name": cifil})
-        files.extend([cifil])
-        if display:
+
+        u0temp, h0temp, u1temp = self._read_files(stage, files)
+        firstm, firstn, dzq = self._calculate_waqua_values(u0temp, h0temp, u1temp)
+            
+        return dzq, firstm, firstn
+    
+    def _search_files(self, stage: int, discharge_value: float) -> Tuple[bool, list]:
+        cblok = str(stage)
+        if self.display:
+            ApplicationSettingsHelper.log_text("input_xyz", dict={"stage": stage, "q": discharge_value})
+            ApplicationSettingsHelper.log_text("---")
```

Comment:
should this not be more clear in a separate method? Like : 
```
def _initialize_file_reading(self, stage, discharge_value):
     if self.display:
            ApplicationSettingsHelper.log_text("input_xyz", dict={"stage": stage, "q": discharge_value})
            ApplicationSettingsHelper.log_text("---")
            ApplicationSettingsHelper.log_text("")
```

Reply:
done